pkg/controller/machine.go

```diff
@@ -832,8 +831,7 @@ func (c *controller) updateMachineStatus(
 	lastOperation v1alpha1.LastOperation,
 	currentStatus v1alpha1.CurrentStatus,
 ) (*v1alpha1.Machine, error) {
-	// Get the latest version of the machine so that we can avoid conflicts
-	latestMachine, err := c.controlMachineClient.Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{})
+	latestMachine, err := c.machineLister.Machines(machine.Namespace).Get(machine.Name)
```

Comment:
I don't think this change wouldn't be enough. Because then the machine update failure would go into an infinite loop and block the MCM. Refer -  https://github.com/gardener/machine-controller-manager/pull/558#issuecomment-725229214.

Reply:
Update: we discussed offline, this cache should update later, and will not fall into an infinite loop. 
Although making many changes in the in-tree machine-controller seems unnecessary as it's already fixed well in OOT, we decided to make the change at one place in reconciliation which is the only major contributor to the API-calls.